pkg/piecestore/psserver/retrieve.go

```diff
@@ -35,17 +35,19 @@ func (s *Server) Retrieve(stream pb.PieceStoreRoutes_RetrieveServer) (err error)
 		return RetrieveError.New("error receiving piece data")
 	}
 
-	authorization := recv.GetAuthorization()
-	if err := s.verifier(authorization); err != nil {
-		return ServerError.Wrap(err)
-	}
-
 	pd := recv.GetPieceData()
 	if pd == nil {
 		return RetrieveError.New("PieceStore message is nil")
 	}
 
-	id, err := getNamespacedPieceID([]byte(pd.GetId()), getNamespace(authorization))
+	rba := recv.GetBandwidthAllocation()
+	if rba == nil {
+		return RetrieveError.New("RenterBandwidthAllocation message is nil")
+	}
+
+	pba := rba.PayerAllocation
```

Comment:
Check `pba` for nil too

Reply:
pba can't be type nil for some reason. Should I check if it's just an empty instance like `pb.PayerBandwidthAllocation{}`?

Reply:
Ah you are right it cannot be nil. Checking if its empty will be ok.